src/components/Proposals/ProposalCard.js

```diff
@@ -41,7 +45,13 @@ function getAttributes(status, theme) {
 
 function ProposalCard({ vote, onProposalClick }) {
   const theme = useTheme()
-  const { context, voteId, description, target } = vote
+  const { context, voteId, script } = vote
+  const {
+    description,
+    emptyScript,
+    targetApp,
+    loading: descriptionLoading,
+  } = useDescribeVote(script, vote.id)
```

Comment:
This hook also returns the target data for the app badge, this feels a bit cleaner to me and we can always pull it out in the future if we need to use it in other contexts.